src/ReferenceIntegrity/Mapping/Driver/Attribute.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace Gedmo\ReferenceIntegrity\Mapping\Driver;
+
+use Doctrine\ORM\Mapping\Driver\AttributeReader;
+use Gedmo\Exception\InvalidMappingException;
+use Gedmo\Mapping\Driver\AbstractAnnotationDriver;
+use Gedmo\ReferenceIntegrity\Mapping\Validator;
+
+/**
+ * This is an attribute mapping driver for ReferenceIntegrity
+ * behavioral extension. Used for extraction of extended
+ * metadata from Attributes specifically for ReferenceIntegrity
+ * extension.
+ *
+ * @author Evert Harmeling <evert.harmeling@freshheads.com>
+ * @license MIT License (http://www.opensource.org/licenses/mit-license.php)
+ */
+class Attribute extends AbstractAnnotationDriver
+{
+    /**
+     * Annotation to identify the fields which manages the reference integrity
+     */
+    const REFERENCE_INTEGRITY = 'Gedmo\\Mapping\\Annotation\\ReferenceIntegrity';
+
+    /**
+     * ReferenceIntegrityAction extension annotation
+     */
+    const ACTION = 'Gedmo\\Mapping\\Annotation\\ReferenceIntegrityAction';
+
+    public function __construct()
+    {
+        $this->setAnnotationReader(new AttributeReader());
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function readExtendedMetadata($meta, array &$config)
```

Comment:
This appears to be copy pasted from `\Gedmo\ReferenceIntegrity\Mapping\Driver\Annotation`. Should it maybe extend the Annotation driver and overwrite the `AnnotationReader`? Or perhaps decorate it?